packages/cmf/src/sagas/http.js

```diff
@@ -213,18 +235,64 @@ export function* httpGet(url, config, options) {
 	return yield* wrapFetch(url, config, HTTP_METHODS.GET, undefined, options);
 }
 
-export const handleDefaultConfiguration = curry((defaultConfig, config) =>
-	mergeCSRFToken(defaultConfig)(config),
+/**
+ * setDefaultHeader - define a default config to use with the saga http
+ * this default config is stored in this module for the whole application
+ *
+ * @param  {object} config key/value of header to apply
+ * @example
+ * import { setDefaultHeader } from '@talend/react-cmf/sagas/http';
+ * setDefaultHeader({headers: {
+ *  'Accept-Language': preferredLanguage,
+ * }});
+ */
+export function setDefaultConfig(config) {
+	if (defaultConfig) {
+		// eslint-disable-next-line no-console
+		console.error(
+			'ERROR: setDefaultConfig should not be called twice, if you wish to change the language use setDefaultLanguage api.',
+		);
+	}
+
+	defaultConfig = config;
+}
+
+/**
+ * To change only the Accept-Language default headers
+ * on the global http defaultConfig
+ * @param {String} language
+ */
+export function setDefaultLanguage(language) {
+	if (get(defaultConfig, 'headers')) {
+		defaultConfig.headers['Accept-Language'] = language;
+	} else {
+		// eslint-disable-next-line no-console
+		console.error('ERROR: you should call setDefaultConfig');
+	}
+}
+
+export const handleDefaultHttpConfiguration = curry((defaultHttpConfig, httpConfig) =>
+	merge(defaultHttpConfig, httpConfig),
 );
 
+/**
+ * getDefaultConfig - return the defaultConfig
+ *
+ * @return {object}  the defaultConfig used by cmf
+ */
+export function getDefaultConfig() {
+	return defaultConfig;
+}
+
 export default {
 	delete: httpDelete,
 	get: httpGet,
 	post: httpPost,
 	put: httpPut,
 	patch: httpPatch,
-	create(defaultConfig = {}) {
-		const configEnhancer = handleDefaultConfiguration(defaultConfig);
+	create(createConfig = {}) {
```

Comment:
`create` should always return a new instance of `http`. If not how I'm I supposed to get a new instance without the default configuration. It also look like the default configuration can be set from multiple point which can be misleading.

Reply:
It is wanted by @jsomsanith  and @jmfrancois. So, if you want an new implementation, talk to you & let's free to change it. Now, I wish to merge this f****** PR.

Reply:
I don't have other solution that use a default configuration if a want use the same headers in many dependencies. So, what do you propose other else ?